radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/CreateTokenToParticleGroupsMapper.java

```diff
@@ -57,19 +61,55 @@
 			null
 		);
 
+		UnallocatedTokensParticle unallocated = new UnallocatedTokensParticle(
+			UInt256.MAX_VALUE,
+			TokenUnitConversions.unitsToSubunits(tokenCreation.getGranularity()),
+			tokenCreation.getAddress(),
+			System.currentTimeMillis(),
+			token.getTokenDefinitionReference(),
+			System.currentTimeMillis() / 60000L + 60000
+		);
+
 		if (tokenCreation.getInitialSupply().compareTo(BigDecimal.ZERO) == 0) {
 			// No initial supply -> just the token particle
-			return Observable.just(ParticleGroup.of(SpunParticle.up(token)));
+			return Observable.just(
+				ParticleGroup.of(SpunParticle.up(token)),
+				ParticleGroup.of(SpunParticle.up(unallocated))
+			);
 		}
 
 		MintedTokensParticle minted = new MintedTokensParticle(
-				TokenUnitConversions.unitsToSubunits(tokenCreation.getInitialSupply()),
+			TokenUnitConversions.unitsToSubunits(tokenCreation.getInitialSupply()),
+			TokenUnitConversions.unitsToSubunits(tokenCreation.getGranularity()),
+			tokenCreation.getAddress(),
+			System.currentTimeMillis(),
+			token.getTokenDefinitionReference(),
+			System.currentTimeMillis() / 60000L + 60000
+		);
+
+		ParticleGroupBuilder mintGroupBuilder = ParticleGroup.builder()
+			.addParticle(unallocated, Spin.DOWN)
+			.addParticle(minted, Spin.UP);
+
+		final UInt256 leftOver = UInt256.MAX_VALUE.subtract(TokenUnitConversions.unitsToSubunits(tokenCreation.getInitialSupply()));
```

Comment:
Just to note, we will definitely want to expose the total supply as a property of the CreateTokenAction - it can just default to UInt256.MAX_VALUE if not set. We can probably just add that as part of this PR, sounds trivial enough.